### PR TITLE
updated Blob constructor to use optional

### LINKED ISSUE #10779: Use `optional` for the Blob constructor
Blocks #2185.

To match https://github.com/w3c/FileAPI/issues/33

* Spec: <https://w3c.github.io/FileAPI/#blob>.
* Code: `components/script/dom/blob.rs`, `components/script/dom/webidls/Blob.webidl`.
* Test: `FileAPI/blob/Blob-constructor.html`.

### PR DESCRIPTION
**Pull request by yoava333**
fixes https://github.com/servo/servo/issues/10779

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10795)

### CODE REVIEW

File: components/script/dom/blob.rs [hunk 1]
```diff
@@ -117,29 +117,27 @@ impl Blob {
...
     // https://w3c.github.io/FileAPI/#constructorBlob
-    pub fn Constructor(global: GlobalRef) -> Fallible<Root<Blob>> {
-        Ok(Blob::new(global, Vec::new(), ""))
-    }
-
-    // https://w3c.github.io/FileAPI/#constructorBlob
-    pub fn Constructor_(global: GlobalRef,
-                        blobParts: Vec<BlobOrString>,
+    pub fn Constructor(global: GlobalRef,
+                        blobParts: Option<Vec<BlobOrString>>,
```
Comment (KiChjang) [inline components/script/dom/blob.rs]: nit: Incorrect indentation.

File: components/script/dom/blob.rs [hunk 2]
```diff
@@ -117,29 +117,27 @@ impl Blob {
...
-                                        },
-                                        &BlobOrString::Blob(ref b) => {
-                                            b.get_data().get_bytes().to_vec()
-                                        },
-                                    }
-                                })
-                                .collect();
+        let bytes: Vec<u8> = match blobParts {
+            None => Vec::new(),
+            Some(blobs) => blobs.iter().flat_map(|bPart| {
```
Comment (Ms2ger) [inline components/script/dom/blob.rs]: Nit: wrap this into `{}`

**Comment by Ms2ger** [comment]
Thanks!

@bors-servo r+

**Comment by jdm** [comment]
@bors-servo: retry
* git failure

**Pull request merged**